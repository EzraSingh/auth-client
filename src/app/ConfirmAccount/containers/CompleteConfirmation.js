import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Redirect } from 'react-router-dom'
import { toast } from 'react-toastify'
import api from 'api'
import Loader from 'app/components/Loader'

class CompleteConfirmation extends Component{
    state = { isLoading: true, complete: false }
    static propTypes = {
        token: PropTypes.string.isRequired
    }

    componentDidMount(){
        api.post("/validate/confirmation", this.props)
        .then((res) => {
            if(res.status === 200){
                toast.success('Registration complete')
                this.setState({ isLoading: false, complete: true })
            }
        })
        .catch((err) => {
            if(err.status === 400){
                this.setState({ isLoading: false, complete: false })
            }
        })
    }

    render(){
        const { isLoading, complete } = this.state
        if(isLoading) return <Loader/>
        if(!isLoading && complete) return <Redirect to="/"/>
        return(
            <section className="container">
                <h3>Invalid Token</h3>
                <p>The confirmation token provided is either <i>invalid</i> or <i>expired</i></p>
            </section>
        )
    }
}

export default CompleteConfirmation