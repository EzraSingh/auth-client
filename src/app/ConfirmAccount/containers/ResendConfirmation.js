import React, { Component } from 'react'
import { Redirect } from 'react-router-dom'
import { toast } from 'react-toastify'
import api from 'api'
import EmailForm from '../components/EmailForm'

class ResendConfirmation extends Component{
    state = { complete : false }
    sendRecoveryLink = event => {
        event.preventDefault()
        const data = new FormData(event.target)
        api.patch("/auth", { email : data.get("email") })
        .then((res) => {
            if(res.status === 200){
                toast.success("New confirmation email sent")
                this.setState({ complete: true })
            }
        })
        .catch((err) => {
            const { message } = err.response.data
            if(err.response.status === 401){              
                toast.warn("No user with that email")
            }
            if(err.response.status === 400){
                toast.error(message)
            }
        })
    }
    render(){
        if(this.state.complete) return <Redirect to="/"/> 
        return <EmailForm handleSubmit={this.sendRecoveryLink}/>
    }
}

export default ResendConfirmation