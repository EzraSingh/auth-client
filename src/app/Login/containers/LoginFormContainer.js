import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Redirect } from 'react-router-dom'
import { connect } from 'react-redux'
import { login } from 'actions/auth'
import LoginForm from '../components/LoginForm'

class LoginFormContainer extends Component{
    static propTypes = { 
        redirect: PropTypes.bool.isRequired,
        login: PropTypes.func.isRequired
    }
    handleSubmit = event => {
        event.preventDefault()
        const data = new FormData(event.target)
        this.props.login({ 
            email : data.get('email'),
            password : data.get('password')
        })
    }
    render(){
        if(this.props.redirect) return <Redirect to='/profile'/>
        return <LoginForm handleSubmit={this.handleSubmit}/>
    }
}

const mapStateToProps = state => {
    return { redirect: state.auth.isLoggedIn }
}

const mapDispatchToProps = dispatch => {
    return { login: credentials => dispatch(login(credentials)) }
}

export { LoginFormContainer }
export default connect(mapStateToProps, mapDispatchToProps)(LoginFormContainer)