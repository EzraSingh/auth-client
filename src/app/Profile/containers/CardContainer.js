import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Loader from 'utils/Loader'
import Card from '../components/Card'
import { connect } from 'react-redux'
import { profile } from 'actions/user'

class CardContainer extends Component{
    static propTypes = {
        isLoading: PropTypes.bool.isRequired,
        email: PropTypes.string,
        profile: PropTypes.object,
        loadProfile: PropTypes.func.isRequired
    }
    componentWillMount(){
        this.props.loadProfile()
    }
    /** Check if profile state has updated */
    shouldComponentUpdate(nextProps){
        return !!nextProps.email ^ this.props.email
    }
    render(){
        const { isLoading, email, profile } = this.props
        return isLoading ? <Loader/> : <Card {...{email, profile}}/>
    }
}

const mapStateToProps = state => {
    return { ...state.user, isLoading: !state.user.email }
}

const mapDispatchToProps = dispatch => {
    return { loadProfile: () => dispatch(profile()) }
}

export { CardContainer }
export default connect(mapStateToProps, mapDispatchToProps)(CardContainer)