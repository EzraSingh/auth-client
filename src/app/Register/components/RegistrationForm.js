import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

const RegistrationForm = ({ handleSubmit }) => {
    return(
        <form onSubmit={handleSubmit}>
            <header>
                <h3>Register</h3>
                <p>Create a new user</p>
            </header>
            <label htmlFor="email">Email</label>
            <input name="email" id="email" type="email" required/>

            <label>Name</label>
            <div className="is-flex">
                <input name="first_name" placeholder="First" required/>
                <input name="last_name" placeholder="Last" required/>
            </div>

            <label htmlFor="password">Password</label>
            <input name="password" id="password" type="password" required/>
            
            <label htmlFor="confirm">Confirm Password</label>
            <input name="confirm" id="confirm" type="password" required/>

            <input type="submit" className="btn" value="Create Account"/>

            <div className="prompt">
                <Link className="link" to='/'>Already have an account?</Link>
                <br/>
                <Link className="link" to="/confirm">Need to resend confirmation link?</Link>
            </div>
        </form>
    )
}

RegistrationForm.propTypes = {
    handleSubmit: PropTypes.func.isRequired
}

export default RegistrationForm