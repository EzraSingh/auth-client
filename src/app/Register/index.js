import React from 'react'
import Form from './components/Form'

const Register = () => {
    return(
        <section id="register" className="view">
            <div className="container animated fadeIn">
                <Form/>
            </div>
        </section>
    )
}

export default Register