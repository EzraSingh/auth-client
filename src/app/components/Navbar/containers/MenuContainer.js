import React, { Component } from 'react'
import { connect } from 'react-redux'
import Menu from '../components/Menu'

/**
 * Route template for registering routes within Navbar
 * @param {} name - Nav item title 
 * @param {} href - Nav link
 * @param {} auth - display only for logged in users
 * @param {} hideOnAuth - display only for anonymous users
 */
function Route(name, href, auth, hideOnAuth){
    this.name = name
    this.href = href
    this.auth = auth
    this.hideOnAuth = hideOnAuth
}

class MenuContainer extends Component{
    static routes = [
        new Route('Profile', '/profile', { auth: true }),
        new Route('Logout', '/logout', { auth: true })
    ]
    /** Filter routes that require authentication */
    generateRoutes(){
        if(!this.props.isLoggedIn){
            // Filter routes that do not require auth
            return MenuContainer.routes.filter((route) => !route.auth)
        }
        else{
            // Filter routes that require auth and are not flagged to hide on auth
            return MenuContainer.routes.filter((route) => route.auth && !route.hideOnAuth)
        }
    }
    render(){
        return <Menu routes={this.generateRoutes()} {...this.props}/>
    }
}

const mapStateToProps = state => {
    return { 
        ...state.auth, 
        profile: {
            avatar: state.user.profile.avatar,
            name: state.user.profile.name
        }
    }
}

export { MenuContainer }
export default connect(mapStateToProps)(MenuContainer)