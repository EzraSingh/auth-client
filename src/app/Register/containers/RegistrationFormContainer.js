import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Redirect } from 'react-router-dom'
import { connect } from 'react-redux'
import { register } from 'actions/user'
import { titleCase } from 'utils/text'
import RegistrationForm from '../components/RegistrationForm'

class RegistrationFormContainer extends Component{
    static propTypes = {
        redirect: PropTypes.bool.isRequired,
        register: PropTypes.func.isRequired
    }
    handleSubmit = (event) => {
        event.preventDefault()
        const data = new FormData(event.target)
        const name = data.get('first_name') + ' ' + data.get('last_name')
        this.props.register({
            email : data.get('email'),
            name: titleCase(name),
            password : data.get('password'),
            confirm : data.get('confirm')
        })
    }
    render(){
        if(this.props.redirect) return <Redirect exact to='/'/>
        return <RegistrationForm handleSubmit={this.handleSubmit}/>
    }
}

const mapStateToProps = state => {
    // NOTE: user.email when null indicates user has not been created
    return { redirect : state.user.email === false }
}

const mapDispatchToProps = dispatch => {
    return { register : (credentials) => dispatch(register(credentials)) }
}

export { RegistrationFormContainer, titleCase }
export default connect(mapStateToProps, mapDispatchToProps)(RegistrationFormContainer)