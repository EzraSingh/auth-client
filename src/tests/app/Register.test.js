import React from 'react'
import { shallow } from 'enzyme'
import Register from 'app/Register'
import { RegistrationFormContainer } from 'app/Register/containers/RegistrationFormContainer'
import RegistrationForm from 'app/Register/components/RegistrationForm'

describe('register view', () => {
    describe('<Register/>', () => {
        it('should render', () => {
            const component = shallow(<Register/>)
            expect(component).toMatchSnapshot()
        })
    })
    describe('<RegistrationFormContainer />', () => {
        const props = { redirect: false, register: jest.fn() }
        const event = { preventDefault: jest.fn() }
        it('should send user registration on form submission', () => {
            const component = shallow(<RegistrationFormContainer {...props}/>)
            component.dive().find('form').simulate('submit', event)
            expect(event.preventDefault).toHaveBeenCalledTimes(1)
            expect(props.register).toHaveBeenCalledTimes(1)
        })
    })
    describe('<RegistrationForm />', () => {
        const props = { handleSubmit: jest.fn() }
        it('should handle form submission', () => {
            const component = shallow(<RegistrationForm {...props}/>)
            component.find('form').simulate('submit')
            expect(props.handleSubmit).toHaveBeenCalledTimes(1)
        })
    })
})