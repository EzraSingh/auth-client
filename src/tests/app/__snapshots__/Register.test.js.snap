// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register view <Register/> should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container animated fadeIn"
      >
        <Connect(RegisterForm) />
      </div>,
      "className": "view",
      "id": "register",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Connect(RegisterForm) />,
        "className": "container animated fadeIn",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container animated fadeIn"
        >
          <Connect(RegisterForm) />
        </div>,
        "className": "view",
        "id": "register",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(RegisterForm) />,
          "className": "container animated fadeIn",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
